File: app/src/main/java/com/coursedicoding/ngajaruiadapter/MainActivity.java
/*************************************************
 * Created by Efendi Hariyadi on 10/10/18 11:25 PM
 * Copyright (c) 2018 . All rights reserved.
 * Last modified 10/10/18 10:05 PM
 ************************************************/

package com.coursedicoding.ngajaruiadapter;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.Toast;

import com.coursedicoding.ngajaruiadapter.adapter.TopBestEmployeeAdapter;
import com.coursedicoding.ngajaruiadapter.model.Employee;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    Integer image[] = {R.drawable.ic_people,R.drawable.ic_people,R.drawable.ic_people,R.drawable.ic_people,R.drawable.ic_people};
    String namaLengkap[] = {"Alfredo Sinaga ","Wawanku","Ade Capri","Axel","Bagas Raga"};
    String phoneNum[] = {"081232","082524","0812345","0811234","086241"};

    float vote[] = new float[] {5, 4, 4, 3, 5};
    int[] umur = new int[] {23,21,31,24,22};
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ////Init RecyclerView Top BestEmployee
        RecyclerView recyclerTopEmp = findViewById(R.id.rv_topbestemployee);
        ArrayList<Employee> beanClassForDashboards = new ArrayList<>();

        for (int i = 0; i < image.length; i++) {
            Employee beanClassForRecyclerView_contacts;
            beanClassForRecyclerView_contacts = new Employee(namaLengkap[i],phoneNum[i],umur[i],image[i],vote[i]);
            beanClassForDashboards.add(beanClassForRecyclerView_contacts);
        }

        RecyclerView.LayoutManager mLayoutManager;
        mLayoutManager = new LinearLayoutManager(MainActivity.this, LinearLayoutManager.HORIZONTAL, false);
        recyclerTopEmp.setLayoutManager(mLayoutManager);
        recyclerTopEmp.setItemAnimator(new DefaultItemAnimator());
        recyclerTopEmp.setAdapter(new TopBestEmployeeAdapter(beanClassForDashboards, new TopBestEmployeeAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(Employee item) {
                Toast.makeText(MainActivity.this, item.getNamaLengkap(), Toast.LENGTH_LONG).show();
            }
        }));
        //end init RecyclerView view

    }
}

File: app/src/main/java/com/coursedicoding/ngajaruiadapter/MainActivity.kt
/*************************************************
 * Created by Efendi Hariyadi on 10/10/18 11:25 PM
 * Copyright (c) 2018 . All rights reserved.
 * Last modified 10/10/18 10:05 PM
 */

package com.coursedicoding.ngajaruiadapter

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.widget.Toast

import com.coursedicoding.ngajaruiadapter.adapter.TopBestEmployeeAdapter
import com.coursedicoding.ngajaruiadapter.model.ModelEmployee


import java.util.ArrayList

class MainActivity : AppCompatActivity() {

    private var image = arrayOf(R.drawable.ic_people, R.drawable.ic_people, R.drawable.ic_people, R.drawable.ic_people, R.drawable.ic_people)
    private var namaLengkap = arrayOf("Alfredo Sinaga ", "Wawanku", "Ade Capri", "Axel", "Bagas Raga")
    private var phoneNum = arrayOf("081232", "082524", "0812345", "0811234", "086241")

    private var vote = floatArrayOf(5f, 4f, 4f, 3f, 5f)
    private var umur = intArrayOf(23, 21, 31, 24, 22)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        ////Init RecyclerView Top BestEmployee
        val recyclerTopEmp = findViewById<RecyclerView>(R.id.rv_topbestemployee)
        val arrayEmployee = ArrayList<ModelEmployee.Employee>()

        for (i in image.indices) {
            val fiilEmployee: ModelEmployee.Employee = ModelEmployee.Employee(namaLengkap[i], phoneNum[i], umur[i], image[i], vote[i], null)
            arrayEmployee.add(fiilEmployee)
        }

        val mLayoutManager: RecyclerView.LayoutManager
        mLayoutManager = LinearLayoutManager(this@MainActivity, LinearLayoutManager.HORIZONTAL, false)
        recyclerTopEmp.layoutManager = mLayoutManager
        recyclerTopEmp.itemAnimator = DefaultItemAnimator()
        recyclerTopEmp.adapter = TopBestEmployeeAdapter(arrayEmployee, object : TopBestEmployeeAdapter.OnItemClickListener {
            override fun onItemClick(item: ModelEmployee.Employee) {
                Toast.makeText(this@MainActivity, item.namaLengkap, Toast.LENGTH_LONG).show()
            }
        })
        //end init RecyclerView view

    }
}
